internal/pkg/docker/dockercompose/volume.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	compose "github.com/compose-spec/compose-go/types"
+	"github.com/docker/go-units"
+	"github.com/dustin/go-humanize/english"
+	"strings"
+)
+
+// The default size that Copilot reserves for a service's ephemeral on-disk storage.
+const defaultEphemeralGiB = 20
+const defaultEphemeralBytes uint64 = defaultEphemeralGiB * units.GiB
+
+type volumeConverter struct {
+	otherSvcVols  map[string][]string
+	copilotVols   map[string]*manifest.Volume
+	topLevelVols  compose.Volumes
+	tmpfsVolNames map[string]bool
+}
+
+func newVolumeConverter(topLevelVols compose.Volumes) volumeConverter {
+	return volumeConverter{
+		otherSvcVols:  map[string][]string{},
+		copilotVols:   map[string]*manifest.Volume{},
+		topLevelVols:  topLevelVols,
+		tmpfsVolNames: map[string]bool{},
+	}
+}
+
+// convertVolumes converts a list of Compose volumes into the Copilot storage equivalent.
```

Comment:
Q: Does this handle both short syntax and long syntax?

Reply:
Yes! Compose converts short syntax to long syntax in almost all places, and my test cases use both.